Google.PowerShell/BigQuery/GcbqDataset.cs

```diff
@@ -0,0 +1,141 @@
+    /// </para>
+    /// <para type="description">
+    /// Creates a new empty dataset in the specified project from a Dataset Object.  This Dataset can be supplied by object via the 
+    /// pipeline or with the -ByObject parameter, or they can be supplied by value with the flags listed below.  If no Project is 
+    /// specified, the default project will be used.  This cmdlet returns a Dataset object.  Required value parameters are to be 
+    /// passed in order after the command, and optional parameters are passed as flags.
+    ///    ByValue Parameters: 
+    /// DatasetId (required) - unique identifier for the dataset.
+    /// Name - descriptive name for the dataset.
+    /// Description - user-friendly description of the dataset.
+    /// Timeout - default duration in ms for tables in the dataset to exist. 
+    /// </para>
+    /// <example>
+    ///   <code>PS C:\> $dataset | New-GcbqDataset -Project "my-project" </code>
+    ///   <para>This command will take a Dataset object from the pipeline and insert it into the Cloud project "my-project". 
+    ///   project $projId</para>
+    /// </example>
+    /// <example>
+    ///   <code>PS C:\> New-GcbqDataset "test_data_id" -Name "Testdata" -Description "Some interesting data!" -Timeout 86400000</code>
+    ///   <para>This command will build a new dataset with the supplied datasetId, name, description, and a timeout of 1 day.</para>
+    /// </example>
+    /// <para type="link" uri="(https://cloud.google.com/bigquery/docs/reference/rest/v2/datasets)">
+    /// [BigQuery Datasets]
+    /// </para>
+    /// </summary>
+    [Cmdlet(VerbsCommon.New, "GcbqDataset")]
+    public class NewGcbqDataset : GcbqCmdlet
+    {
+        private class ParameterSetNames
+        {
+            public const string ByObject = "ByObject";
+            public const string ByValues = "ByValue";
+        }
+
+        /// <summary>
+        /// <para type="description">
+        /// The project to look for datasets in. If not set via PowerShell parameter processing, will
+        /// default to the Cloud SDK's DefaultProject property.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false, ParameterSetName = ParameterSetNames.ByObject)]
+        [Parameter(Mandatory = false, ParameterSetName = ParameterSetNames.ByValues)]
+        [ConfigPropertyName(CloudSdkSettings.CommonProperties.Project)]
+        public string Project { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The dataset object that will be sent to the server to be inserted.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false, ValueFromPipeline = true, ParameterSetName = ParameterSetNames.ByObject)]
+        public Dataset Dataset { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The DatasetId for the datset reference object. Must be unique within the project, and must be 1024 
```

Comment:
Consider just saying "Must be unique within the project". The 1 to 1024 characters is kinda of superfluous. (Since I highly doubt people will run into that 1024 limit and be surprised.)